csrc/host_ir/lower.h

```diff
@@ -14,22 +14,30 @@
 
 namespace nvfuser {
 
+struct HostIrLowerParams {
+  CommunicatorBackend communicator_backend = CommunicatorBackend::kNccl;
+};
+
 class HostIrLower {
  public:
+  HostIrLower(HostIrLowerParams params = HostIrLowerParams())
+      : params_(params) {}
+
   // The flag `ignore_inner_resharding` is useful because the preseg passes
   // `InsertReshardingsPass` and `ReorderShardedAxisPass` want different
   // behaviors
   static bool canLower(Expr* expr, bool ignore_inner_resharding = false);
 
   // Lower a sharded Expr into a series of Communication.
-  static std::vector<Expr*> lower(Expr* c);
+  std::vector<Expr*> lower(Expr* c);
 
-  static std::unique_ptr<hir::HostIrContainer> lower(
+  std::unique_ptr<hir::HostIrContainer> lower(
       std::unique_ptr<Fusion> fusion,
       int64_t my_device_index);
 
  private:
-  static std::vector<Expr*> lowerToCollectiveBasedPipelinedGemmComm(Expr* expr);
+  std::vector<Expr*> lowerToCollectiveBasedPipelinedGemmComm(Expr* expr);
+  HostIrLowerParams params_;
```

Comment:
```suggestion
  const HostIrLowerParams params_;
```
This way, readers know params_ won't be changed after construction.